### ISSUE TITLE
SVG: "RangeError: Invalid string length"

### ISSUE THREAD

**Issue by doublex** [mozilla/pdf.js#8434]
Steps to reproduce (attached file)
[invalidstringlength.pdf](https://github.com/mozilla/pdf.js/files/1021334/invalidstringlength.pdf)

````
npm install
gulp dist
node examples/node/pdf2svg.js /tmp/invalidstringlength.pdf
````

Console:
````
# Document Loaded
Number of Pages: 1

# Page 1
Size: 595.32x841.92

Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented graphic state BM
Warning: Unimplemented graphic state ca
Warning: Unimplemented graphic state BM
Warning: Unimplemented graphic state ca
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Warning: Unimplemented operator dependency
Error: RangeError: Invalid string length
````

**Comment by doublex** [comment]
$50 bounty:
https://www.bountysource.com/issues/45475031-svg-rangeerror-invalid-string-length

**Comment by doublex** [comment]
$50 bounty:
https://www.bountysource.com/issues/45475031-svg-rangeerror-invalid-string-length

**Comment by Rob--W** [comment]
The issue is caused by the fact that the generated SVG file is too large, it exceeds the maximum length that V8 allows ([256M](https://chromium.googlesource.com/v8/v8.git/+/13d38e4a87153ab3b0a8d0612ea1c7ee3f664d77/src/objects/string.h#346)).
I'll fix this once #8540 is merged, by adding the ability to stream the generated SVG data instead of generating everything at once.
